win32ss/user/winsrv/consrv/condrv/text.c

```diff
@@ -977,6 +977,18 @@ IntWriteConsoleOutputStringUnicode(
         if (bCJK && Ptr->Char.UnicodeChar >= 0x80 &&
             mk_wcwidth_cjk(Ptr->Char.UnicodeChar) == 2)
         {
+            if (X == Buffer->ScreenBufferSize.X - 1)
+            {
+                /* new line */
+                X = 0;
+                ++Y;
+                if (Y == Buffer->ScreenBufferSize.Y)
```

Comment:
Now that I read again this part of the code (for Y) I'm skeptical.
I would ask you to do the following test (both on Windows and on ROS with your patch):
What happens if you resize the console screen buffer (from the console properties dialog) to the standard 80x25 size,
then manage to fill all the lines with double-width characters, and on the very last line, you also add a single-width character (say, at its beginning) such that the last double-width character of this last line is at position `Buffer->ScreenBufferSize.Y - 3 == 78` (so that it spans cells 77 and 78 -- cells are indexed from 0) , and **ONLY** then you add another double-width character ?